syne_tune/optimizer/schedulers/neuralbands/neuralband_supplement.py

```diff
@@ -0,0 +1,494 @@
+                    largest_update_resource = self._active_trials[trial_id][
+                        "largest_update_resource"
+                    ]
+                    if largest_update_resource is None:
+                        largest_update_resource = resource - 1
+                    assert largest_update_resource <= resource, (
+                        f"Internal error (trial_id {trial_id}): "
+                        + f"on_trial_result called with resource = {resource}, "
+                        + f"but largest_update_resource = {largest_update_resource}"
+                    )
+                    if resource == largest_update_resource:
+                        do_update = False  # Do not update again
+                    else:
+                        self._active_trials[trial_id][
+                            "largest_update_resource"
+                        ] = resource
+                if not task_continues:
+                    if (not self.does_pause_resume()) or resource >= self.max_t:
+                        trial_decision = SchedulerDecision.STOP
+                        act_str = "Terminating"
+                    else:
+                        trial_decision = SchedulerDecision.PAUSE
+                        act_str = "Pausing"
+                    self._cleanup_trial(trial_id)
+                if debug_log is not None:
+                    if not task_continues:
+                        logger.info(
+                            f"trial_id {trial_id}: {act_str} evaluation "
+                            f"at {resource}"
+                        )
+                    elif milestone_reached:
+                        msg = f"trial_id {trial_id}: Reaches {resource}, continues"
+                        next_milestone = task_info.get("next_milestone")
+                        if next_milestone is not None:
+                            msg += f" to {next_milestone}"
+                        logger.info(msg)
+            self.searcher.on_trial_result(
+                trial_id, config, result=result, update=do_update
+            )
+        # Extra info in debug mode
+        log_msg = f"trial_id {trial_id} (metric = {result[self.metric]:.3f}"
+        for k, is_float in ((self._resource_attr, False), ("elapsed_time", True)):
+            if k in result:
+                if is_float:
+                    log_msg += f", {k} = {result[k]:.2f}"
+                else:
+                    log_msg += f", {k} = {result[k]}"
+        log_msg += f"): decision = {trial_decision}"
+        logger.debug(log_msg)
+        return trial_decision
+
+    
+class NeuralbandEGreedyScheduler(NeuralbandSchedulerBase):
+    def __init__(self, config_space, epsilon = 0.1, step_size = 30, max_while_loop = 100,  **kwargs):
+        """
+        We combine the epsilon-greedy strategy with NeuralBand, where we randomly select configurations with 
+        probability epsilon or use the greedy strategy to select configurations with probability 1-epsilon
```

Comment:
'where we randomly select configurations with probability epsilon or use the greedy strategy to select configurations with probability 1-epsilon' -> 'where, with probability epsilon, we select configurations either randomly or, with probability  1 - epsilon, greedily by maximizing the acquisition function'

Reply:
Done.